dali/operators/reader/loader/video/frames_decoder_gpu.h

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_READER_LOADER_VIDEO_FRAMES_DECODER_GPU_H_
+#define DALI_OPERATORS_READER_LOADER_VIDEO_FRAMES_DECODER_GPU_H_
+
+#include "dali/operators/reader/loader/video/frames_decoder.h"
+
+#include <string>
+#include <memory>
+#include <queue>
+#include <vector>
+
+#include "dali/operators/reader/loader/video/nvdecode/cuviddec.h"
+#include "dali/operators/reader/loader/video/nvdecode/nvcuvid.h"
+
+#include "dali/core/dev_buffer.h"
+
+namespace dali {
+struct NvDecodeState {
+  CUvideodecoder decoder_;
+  CUvideoparser parser_;
```

Comment:
```suggestion
  CUvideodecoder decoder;
  CUvideoparser parser;
```
I think the idea is that struct have all fields public so we don't need to use naming as for internal fields.

Reply:
Done